sdk/mobile-center-distribute/src/test/java/com/microsoft/azure/mobile/distribute/DistributeBeforeDownloadTest.java

```diff
@@ -740,7 +743,43 @@ public ServiceCall answer(InvocationOnMock invocation) throws Throwable {
 
         /* Verify no download scheduled. */
         verifyStatic(never());
-        AsyncTaskUtils.execute(anyString(), any(Distribute.DownloadTask.class), Mockito.<Void>anyVararg());
+        AsyncTaskUtils.execute(anyString(), any(DownloadTask.class), Mockito.<Void>anyVararg());
+    }
+
+    @Test
+    public void mandatoryUpdateDialog() throws Exception {
+
+        /* Mock we already have token. */
+        when(PreferencesStorage.getString(PREFERENCE_KEY_UPDATE_TOKEN)).thenReturn("some token");
+        HttpClientNetworkStateHandler httpClient = mock(HttpClientNetworkStateHandler.class);
+        whenNew(HttpClientNetworkStateHandler.class).withAnyArguments().thenReturn(httpClient);
+        when(httpClient.callAsync(anyString(), anyString(), anyMapOf(String.class, String.class), any(HttpClient.CallTemplate.class), any(ServiceCallback.class))).thenAnswer(new Answer<ServiceCall>() {
+
+            @Override
+            public ServiceCall answer(InvocationOnMock invocation) throws Throwable {
+                ((ServiceCallback) invocation.getArguments()[4]).onCallSucceeded("mock");
+                return mock(ServiceCall.class);
+            }
+        });
+        ReleaseDetails releaseDetails = mock(ReleaseDetails.class);
+        when(releaseDetails.getId()).thenReturn(4);
+        when(releaseDetails.getVersion()).thenReturn(7);
+        when(releaseDetails.isMandatoryUpdate()).thenReturn(true);
+        when(ReleaseDetails.parse(anyString())).thenReturn(releaseDetails);
+
+        /* Trigger call. */
+        Distribute.getInstance().onStarted(mContext, "a", mock(Channel.class));
+        Distribute.getInstance().onActivityResumed(mock(Activity.class));
+
+        /* Verify release notes persisted. */
+        verifyStatic();
+        PreferencesStorage.putString(PREFERENCE_KEY_RELEASE_DETAILS, "mock");
+
+        /* Verify dialog. */
+        verify(mDialogBuilder, never()).setNeutralButton(anyString(), any(DialogInterface.OnClickListener.class));
```

Comment:
setNeutral was already covered in the postpone test.
I just added checks to make sure the dialog is cancelable instead.